check gofmt with golang 1.4

eparis opened the pull request.

eparis committed:
check gofmt with golang 1.4

vmarmol commented on pkg/proxy/config/config.go:
```diff
@@ -262,7 +262,7 @@ func (s *serviceStore) MergedState() interface{} {
 // when changes occur.
 func watchForUpdates(bcaster *config.Broadcaster, accessor config.Accessor, updates <-chan struct{}) {
-	for _ = range updates {
+	for range updates {
```
Looks like Go 1.3 doesn't like this from the Travis breakage.

eparis commented:
hmmm, so gofmt 1.4 with -l requires this change, but gofmt from 1.3 in shippable dies.  I have no idea how to handle this.  Just keep ignore people who use 1.4 ?

eparis commented:
we could drop the -l check and then gofmt 1.4 would be happy with the old code....

eparis commented:
-s is the problem, not -l.  whoops

eparis committed:
change proxy to satisfy both gofmt 1.3 and 1.4 gofmt

gofmt -s from 1.4  does not like
	for _ = range BLAH
it wants
	for range BLAH

But gofmt from 1.3 dies:
	./pkg/proxy/config/config.go:265:6: expected operand, found 'range'
	./pkg/proxy/config/config.go:268:3: expected '{', found 'EOF'

So instead, rewrite the code to make them both happy

vmarmol commented:
LGTM, thanks @eparis! Will wait for green for merge
